packages/workspace/src/browser/workspace-service.ts

```diff
@@ -57,18 +67,149 @@ export class WorkspaceService implements FrontendApplicationContribution {
 
     @postConstruct()
     protected async init(): Promise<void> {
-        const rootUri = await this.server.getRoot();
-        this._root = await this.toValidRoot(rootUri);
+        this._workspaceDescription = {
+            root: undefined,
+            roots: [],
+            workspaceConfigFile: undefined
+        };
+        await this.setupRoots();
```

Comment:
the flow still looks complicated with duplicate code, please simplify:
- there should be a method `updateWorkspace`, it should get roots, update current description and fire an event
   - there should not be other places where roots are updated or an event is fired, please check
- it should be called on `init` instead of `setupRoots`
- it should be called on `onFilesChanged` whenever a configuration file is affected regardless of this change